src/clients/K8sClient.ts

```diff
@@ -0,0 +1,10 @@
+
+import * as k8s from '@kubernetes/client-node';
+export class K8sClient {
+    public readonly k8sApi: k8s.CoreV1Api;
+    public constructor() {
+        const kc = new k8s.KubeConfig();
+        kc.loadFromDefault();
```

Comment:
This will read the default config, which I don't think is necessarily the same as the one in `AccountContextManager`.

Rather than creating a new class here, could we add a function to AccountContextManager that returns the k8s API client? Internally that would call `getKubeconfigPathAsync()`, and the client could then be created using `kc.loadFromString(<fileContent>)`.

That would also avoid the extra constructor parameter to `KubectlClient` and `KubectlClientProvider`.

Once everything is using the new client, `getKubeconfigPathAsync` could probably be made private.

What do you think?

Reply:
My idea was to replace or get rid of account context manager which creates dependency on Kubernetes extension. But that is long term, these changes are temporary I think immediately I can do another PR to replace everything with javasript client. Thoughts? Only downside I can think of is if kubernetes extension uses different kube config location from default one and that might cause issues ?

Reply:
> if Kubernetes extension uses different Kube config location from default one

Yep, that's what I'm thinking. The K8s extension has some [non-trivial](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/27835b6ced4607ac69bd66c73f344ee9c878abac/src/components/kubectl/kubeconfig.ts#L51-L85) logic for determining the `kubeconfig` path.

Does B2K depend on the K8s extension for other things, or just this? Is the aim to completely remove the dependency on that extension? I had a feeling some of the commands were launched from the K8s cluster explorer treeview. If that's the case, users would definitely be confused by any discrepancy between the way the two extensions locate the `kubeconfig`.

I agree that replacing `AccountContextManager` with something else would be one migration path towards removing the K8s kubectl API dependency. But another one would be to change the interface by first exposing newer methods, then deprecating/removing older ones and removing the extension dependency. I think that would reduce the impact on code/classes that depend on `AccountContextManager` because they wouldn't have to take new dependencies. Does that sound like a reasonable approach?

Another option would be to go a step further and introduce a type that entirely abstracts away the underlying K8s implementation (whether that's the extension's Kubectl API or the JS client), and have `AccountContextManager` return _that_. Not sure if that'd make sense here, but keeping everything in `AccountContextManager` for now would keep that option open.

Reply:
As discussed updated the code. @peterbom .